index.js

```diff
@@ -877,67 +865,116 @@ class Watcher extends EventEmitter {
     this.opened = false
     this.closed = false
 
-    this.running = false
-
     this.latestDiff = 0
     this.range = range
+
+    this.current = null
+    this.previous = null
     this.stream = null
 
-    this._onappendBound = debounceify(this._onappend.bind(this))
+    this._lock = mutexify()
+    this._resolveOnChange = null
 
     this._opening = this._ready().catch(safetyCatch)
```

Comment:
needs to be

``` js
this._opening = this._ready()
this._opening.catch(safetyCatch)
```

Otherwise you are supressing explicit errors always

Reply:
Wow wow haha, can you explain the difference?

Reply:
Ah due this line? https://github.com/holepunchto/hyperbee/blob/f9f6ffa8ff7e77ea834750ee069fdf2ca89a2d24/index.js#L912

Reply:
Makes sense, at first I didn't even realized the `_opening` usage in `next()`